model/src/pyrenew/arrayutils.py

```diff
@@ -92,3 +94,158 @@ def pad_x_to_match_y(
     return pad_to_match(
         x, y, fill_value=fill_value, pad_direction=pad_direction, fix_y=True
     )[0]
+
+
+class PeriodicProcessSample(NamedTuple):
+    """
+    A container for holding the output from `process.PeriodicProcess.sample()`.
+
+    Attributes
+    ----------
+    value : ArrayLike
+        The sampled quantity.
+    """
+
+    value: ArrayLike | None = None
+
+    def __repr__(self):
+        return f"PeriodicProcessSample(value={self.value})"
+
+
+class PeriodicBroadcaster:
+    r"""
+    Broadcast arrays periodically using either repeat or tile,
+    considering period size and starting point.
+    """
+
+    def __init__(
+        self,
+        data_starts: int,
+        period_size: int,
+        broadcast_type: str,
+    ) -> None:
+        """
+        Default constructor for PeriodicBroadcaster class.
+
+        Parameters
+        ----------
+        data_starts : int
```

Comment:
suggest a more generic name, like `offset`

Reply:
Done in 85b9b00